azure-pipelines.yml

```diff
@@ -63,23 +63,33 @@ stages:
               Debug:
                 _BuildConfig: Debug
         steps:
-        - script: eng\common\cibuild.cmd
-            -configuration $(_BuildConfig)
-            -prepareMachine
-            $(_InternalBuildArgs)
-          name: Build
-          displayName: Build
-          condition: succeeded()
+        - ${{ if eq(variables._RunAsPublic, False) }}:
+          - script: eng\InternalBuild.cmd
+              -configuration $(_BuildConfig)
+              -prepareMachine
+              $(_InternalBuildArgs)
+            name: Build
+            displayName: Build
+            condition: succeeded()
+        - ${{ if eq(variables._RunAsPublic, True) }}:
+          - script: eng\common\CIBuild.cmd
+              -configuration $(_BuildConfig)
+              -prepareMachine
+              $(_InternalBuildArgs)
+            name: Build
+            displayName: Build and test
+            condition: succeeded()
 
-        - task: PublishTestResults@2
-          displayName: 'Publish Unit Test Results'
-          inputs:
-            testResultsFormat: xUnit
-            testResultsFiles: '$(Build.SourcesDirectory)/artifacts/TestResults/**/*.xml'
-            mergeTestResults: true
-            searchFolder: $(system.defaultworkingdirectory)
-            testRunTitle: XHarness unit tests - $(Agent.JobName)
-          condition: succeededOrFailed()
+        - ${{ if eq(variables._RunAsPublic, True) }}:
```

Comment:
This doesn't need to be conditional, there just won't be test results if you don't make them.

Reply:
It gives warnings though:
![image](https://user-images.githubusercontent.com/7013027/80000157-1426e480-84bd-11ea-9181-8757d267591f.png)

Not sure we want those on every build, so I added these guards

Reply:
sure that's fine.  Less pressing than the "don't duplicate the whole script to remove 1 arg" issue :)